framework/global/settings.cpp

```diff
@@ -48,127 +48,122 @@ Settings::~Settings()
     }
+}
 
-    static Item null;
-    return null;
+void Settings::load()
+{
+    m_items = readItems();
 }
 
-Settings::Item& Settings::findItem(const Key& key)
+Settings::Items Settings::readItems() const
 {
-    auto it = m_items.find(key);
-    if (it != m_items.end()) {
-        return it->second;
+    Items result;
+
+    for (const QString& key : m_settings->allKeys()) {
+        Item item;
+        item.key = Key(std::string(), key.toStdString());
+        item.value = Val::fromQVariant(m_settings->value(key));
+
+        result[item.key] = item;
     }
 
-    static Item null;
-    return null;
+    return result;
 }
 
-const std::map<Settings::Key, Settings::Item>& Settings::items() const
+Val Settings::value(const Key& key) const
 {
-    return m_items;
+    return findItem(key).value;
 }
 
-/**
- * @brief Settings::reload method needed only for compatibility with the old MU preferences
- */
-void Settings::reload()
+Val Settings::defaultValue(const Key& key) const
 {
-    std::map<QString, Item& > items;
-    for (auto it = m_items.begin(); it != m_items.end(); ++it) {
-        Item& item = it->second;
-        items.insert({ QString::fromStdString(item.key.key), item });
-    }
+    return findItem(key).defaultValue;
+}
 
-    QStringList keys = m_settings->allKeys();
-    for (const QString& key : keys) {
-        auto it = items.find(key);
-        if (it == items.end()) {
-            continue;
-        }
+void Settings::setValue(const Key& key, const Val& value)
+{
+    writeValue(key, value);
 
-        Item& item = it->second;
+    Item& item = findItem(key);
 
-        QVariant val = m_settings->value(key);
+    if (item.isNull()) {
+        m_items[key] = Item{ key, value, value };
+    } else {
+        item.value = value;
+    }
 
-        setValue(item.key, Val::fromQVariant(val));
+    auto it = m_channels.find(key);
+    if (it != m_channels.end()) {
+        Channel<Val> channel = it->second;
+        channel.send(value);
     }
 }
 
-void Settings::load()
+void Settings::writeValue(const Key& key, const Val& value)
 {
-    std::map<QString, Item& > items;
-    for (auto it = m_items.begin(); it != m_items.end(); ++it) {
-        Item& item = it->second;
-        items.insert({ QString::fromStdString(item.key.key), item });
-    }
-
-    QStringList keys = m_settings->allKeys();
-    for (const QString& key : keys) {
-        auto it = items.find(key);
-        if (it == items.end()) {
-            // LOGW() << "not found item with key: " << key;
-            continue;
-        }
-
-        Item& item = it->second;
-        item.val = Val::fromQVariant(m_settings->value(key));
-        item.val.setType(item.defaultVal.type());
-    }
+    // TODO: implement writing/reading first part of key (module name)
+    m_settings->setValue(QString::fromStdString(key.key), value.toQVariant());
 }
 
-Val Settings::value(const Key& key) const
+void Settings::setDefaultValue(const Key& key, const Val& value)
```

Comment:
needed to send notifications about change default value

Reply:
There is no channel like "defaultValueChanged", so I don't think that need send any notification from this method at the current moment